core/data-layer/src/data_layer.rs

```diff
@@ -296,13 +297,23 @@ impl DataLayer {
             return Err(anyhow!(result.err().unwrap().to_string()));
         }
         let result = result.unwrap();
+
+        let config_mapper_data = get_config_mapper();
+
         for row in result {
+            let metadata_str: Option<&str> = row.try_get("metadata").ok();
+            let metadata = if let Some(meta_str) = metadata_str {
+                execute_config_mapper(meta_str.to_string(), config_mapper_data.clone()).unwrap()
```

Comment:
do not unwrap right away. Add an error handling

Reply:
I modified the statements to use map_err and ? operator.